app/src/main/java/com/example/grutor/Activites/DetailActivity.java

```diff
@@ -1,42 +1,75 @@
 package com.example.grutor.Activites;
 
+import androidx.annotation.RequiresApi;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.ContextCompat;
 import androidx.fragment.app.FragmentManager;
 
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.graphics.Color;
+import android.os.Build;
 import android.os.Bundle;
 import android.view.View;
 import android.widget.AdapterView;
 import android.widget.ArrayAdapter;
+import android.widget.Button;
 import android.widget.Spinner;
+import android.widget.TextView;
 
 import com.example.grutor.R;
 
+import org.w3c.dom.Text;
+
 public class DetailActivity extends AppCompatActivity {
 
     protected Spinner spDayOfWeek;
     protected Spinner spTopicsList;
     protected String desiredSubject = "";
     protected Bundle bundle;
+    protected Button btnConfirm;
+    protected Button btnHw;
+    protected Button btnExam;
+    protected Button btnEssay;
+    protected Button btnOther;
+    protected TextView tvProblem1;
+    protected TextView tvProblem2;
+    protected TextView tvProblem3;
+    protected boolean isColor;
+    public int KEY_BLUEBLACK;
+    public int KEY_BUTTONPRIMARY;
+    public int KEY_BLUEBLACK_LIGHT;
```

Comment:
except they are define to a static constant value. otherwise the other call should not call a variable in an Activity (activity can be destroy anytime).